kagen/context.cpp

```diff
@@ -342,6 +342,12 @@ PGeneratorConfig CreateConfigFromString(const std::string& options_str, PGenerat
     config.coordinates = get_bool_or_default("coordinates");
     config.permute     = get_bool_or_default("permute");
 
+    // edge weight generation
+    const std::string edge_weight_generator_name = get_string_or_default("edgeweight-generator", "none");
+    config.edge_weights.generator_type = GetEdgeWeightGeneratorTypeMap()[edge_weight_generator_name];
+    config.edge_weights.weight_range_begin = get_sint_or_default("edgeweight-range-begin", 1);
```

Comment:
different default value than what is set as default value in `context.h`; maybe set these values from a default initialized `EdgeWeightConfig` object so that there are not two places in the code that should stay in sync?